apollo-ios/Sources/Apollo/GraphQLResult.swift

```diff
@@ -47,3 +47,28 @@ extension GraphQLResult: Equatable where Data: Equatable {
 }
 
 extension GraphQLResult: Hashable where Data: Hashable {}
+
+extension GraphQLResult {
+  public func asDictionary() -> [String: Any] {
+    var dict: [String: Any] = [:]
+    if let data { dict["data"] = convert(value: data.__data) }
+    if let errors { dict["errors"] = errors.map { $0.asDictionary() } }
+    if let extensions { dict["extensions"] = extensions }
+    return dict
+  }
+  
+  private func convert(value: Any) -> Any {
+      var val: Any = value
```

Comment:
@AnthonyMDev @calvincestari Started with essentially what was in the issue with a few adjustments to fix errors. But I'm wondering if it would be better to have `val` be optional and then do a guard/throws before attempting to use it.

Similarly for the 2 `asDictionary` functions, similar for the `asDictionary` functions if there should maybe be some minimum requirements like if the result dictionary is empty, should we throw an error vs returning an empty dictionary to make it more clear to the user that its empty, or is it reasonable that they could have an empty result?

Reply:
IMO there is only value in optional if we're going to allow/return nulls. If we're simply going to `guard` and `throw` there is little benefit in a `private` function. Yes it's more verbose to have to check `data`, `errors`, `extensions` but it's clearer what you're going to get out of `convert`?

Reply:
Yeah I agree with @calvincestari here. I also think it's okay to return an empty dictionary. If the result data were to bee empty, that's what we would want.